tests/ocp/sriov/internal/sriovenv/sriovenv.go

```diff
@@ -0,0 +1,2402 @@
+	if err := waitForSriovPolicy(apiClient, config.ResourceName, config.Namespace); err != nil {
+		return fmt.Errorf("SRIOV policy %q must exist before NAD creation: %w",
+			config.ResourceName, err)
+	}
+
+	// Wait for NAD creation
+	if err := waitForNAD(apiClient, config.Name, config.NetworkNamespace); err != nil {
+		return fmt.Errorf("failed to wait for NAD %q: %w", config.Name, err)
+	}
+
+	// Check VF resources (non-blocking for transient issues, but fail fast on config errors)
+	klog.V(90).Infof("Verifying VF resources for %q", config.ResourceName)
+
+	var lastVFErr error
+
+	err = wait.PollUntilContextTimeout(
+		context.Background(),
+		tsparams.VFResourcePollingInterval,
+		tsparams.VFResourceTimeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			available, verifyErr := VerifyVFResourcesAvailable(
+				apiClient, ocpsriovinittools.SriovOcpConfig, config.ResourceName)
+			if verifyErr != nil {
+				lastVFErr = verifyErr
+				// Check for hard configuration errors that should fail immediately
+				errMsg := verifyErr.Error()
+				if strings.Contains(errMsg, "cannot be nil") ||
+					strings.Contains(errMsg, "worker label is empty") {
+					// Configuration error - fail fast
+					return false, verifyErr
+				}
+				// Transient error (API issues, etc.) - keep retrying
+				klog.V(90).Infof("Transient error checking VF resources: %v", verifyErr)
+
+				return false, nil
+			}
+
+			return available, nil
+		})
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			// Log the last error for debugging if we timed out
+			if lastVFErr != nil {
+				klog.V(4).Infof("VF resources %q not available after timeout. Last error: %v",
+					config.ResourceName, lastVFErr)
+			}
+
+			klog.V(90).Infof("VF resources %q not yet available (may still be provisioning). "+
+				"Will proceed and re-check when pods are created.", config.ResourceName)
+			// Proceed: VF availability will be re-checked by pod creation path
+		} else {
+			return fmt.Errorf("failed to verify VF resources %q: %w", config.ResourceName, err)
+		}
+	}
+
+	klog.V(90).Infof("SRIOV network %q created and ready", config.Name)
+
+	return nil
+}
+
+const (
+	// DefaultMcpLabel is the default MachineConfigPool label for worker nodes.
+	DefaultMcpLabel = "machineconfiguration.openshift.io/role=worker"
+)
+
+// CheckSriovOperatorStatus checks if SR-IOV operator is running and healthy.
+func CheckSriovOperatorStatus(apiClient *clients.Settings, config *sriovconfig.SriovOcpConfig) error {
+	if config == nil {
+		return fmt.Errorf("SriovOcpConfig cannot be nil")
+	}
+
+	klog.V(90).Infof("Checking SR-IOV operator status in namespace %q", config.OcpSriovOperatorNamespace)
+	// Use the centralized function from sriovoperator package
+
+	return sriovoperator.IsSriovDeployed(apiClient, config.OcpSriovOperatorNamespace)
+}
+
+// WaitForSriovPolicyReady waits for SR-IOV policy to be ready and MCP to be stable.
+func WaitForSriovPolicyReady(
```

Comment:
What's the reason for this func. Why don't use WaitForSriovAndMCPStable directly